### PR TITLE
SOC-1099 Fixed problem not getting views digest plus some cleanup

### PR DESCRIPTION
**Pull request by garthwebb**
https://wikia-inc.atlassian.net/browse/SOC-1099

### CODE REVIEW

File: extensions/wikia/Email/Controller/FounderDigestController.class.php [hunk 1]
```diff
@@ -53,8 +53,8 @@ protected function assertValidWikiId() {
 	 * @throws \Email\Check
 	 */
 	protected function assertPageViewsSet() {
```
Comment (garthwebb) [inline extensions/wikia/Email/Controller/FounderDigestController.class.php:55]: This logic was a little off; checking for an empty value but not a stringified zero value.  My assumption was that this was supposed to let zero pages views generate an email, but this won't work for integer zero values.  Cristina and I agree we should just skip all zero page view emails.

File: extensions/wikia/Email/Controller/FounderDigestController.class.php [hunk 2]
```diff
@@ -126,7 +126,12 @@ public function assertCanEmail() {
 	 * @throws \Email\Check
 	 */
 	protected function assertSubscribedToCompleteDigest() {
```
Comment (garthwebb) [inline extensions/wikia/Email/Controller/FounderDigestController.class.php:128]: Updated call here and two other places to the new get*Preference calls

File: extensions/wikia/FounderEmails/FounderEmails.class.php [hunk 3]
```diff
@@ -102,7 +106,7 @@ public function processEvents( $eventType, $useMasterDb = false, $wikiId = null
 			}
 
 			$oEvent = FounderEmailsEvent::newFromType( $eventType );
-			$result = $oEvent->process( $aEventsData );
+			$oEvent->process( $aEventsData );
```
Comment (garthwebb) [inline extensions/wikia/FounderEmails/FounderEmails.class.php:109]: Nothing gets returned by `process`

File: extensions/wikia/FounderEmails/FounderEmails.class.php [hunk 4]
```diff
@@ -4,8 +4,8 @@ class FounderEmails {
 	static private $instance = null;
```
Comment (garthwebb) [inline extensions/wikia/FounderEmails/FounderEmails.class.php:4]: Almost all clean up here, one exception marked (still pretty much clean up though).

File: extensions/wikia/FounderEmails/generate.php [hunk 5]
```diff
@@ -1,164 +0,0 @@
-<?php
```
Comment (garthwebb) [inline extensions/wikia/FounderEmails/generate.php:1]: This is replaced by the Special:Email page.

File: includes/wikia/services/DataMartService.class.php [hunk 6]
```diff
@@ -26,46 +26,44 @@ class DataMartService extends Service {
 	const TTL = 43200; // WikiaSQL results caching time (12 hours)
```
Comment (garthwebb) [inline includes/wikia/services/DataMartService.class.php:26]: Ran stylize.php on this file.  Lots of spacing changes.  Only other changes here are removing wfProfile\* calls and a few variable renames to meet code standards.  I'd recommend adding `?w=1` to look at those.

**Comment by mklucsarits** [comment]
LGTM

File: extensions/wikia/FounderEmails/events/FounderEmailsViewsDigestEvent.class.php [hunk 7]
```diff
@@ -40,14 +38,22 @@ public function process ( Array $events ) {
...
+
+			// Don't bother sending this email if there are no page views for this period
+			if ( empty( $views ) ) {
+				continue;
+			}
+
 			$emailParams = [
-				'pageViews' => $founderEmailObj->getPageViews( $cityID )
+				'pageViews' => $views,
+				'wikiId' => $cityID,
```
Comment (garthwebb) [inline extensions/wikia/FounderEmails/events/FounderEmailsViewsDigestEvent.class.php:50]: This is the real bug of this PR; `wikiId` is required field but it wasn't being passed in.

**Comment by jsutterfield** [comment]
LGTM

**Pull request merged**